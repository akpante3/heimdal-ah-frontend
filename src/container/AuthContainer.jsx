import { Component } from 'react';
// import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import login from '../actions/auth/login';
import validateLoginInput from '../validations/authValidations';

class AuthContainer extends Component {
  state = {
    email: '',
    password: '',
    errors: {},
  };

  onChange = (event) => {
    const { errors } = this.state;
    if (errors[event.target.name]) {
      const newErrors = { ...errors };
      delete newErrors[event.target.name];
      this.setState({
        [event.target.name]: event.target.value,
        errors: newErrors,
      });
    } else {
      this.setState({
        [event.target.name]: event.target.value,
      });
    }
  };

  onLoginSubmit = (e) => {
    const { actions } = this.props;
    e.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      actions.login(this.state);
    }
  };

  isValid = () => {
    const { errors, isValid } = validateLoginInput(this.state);
    if (!isValid) {
      this.setState({ errors, password: '' });
    }
    return isValid;
  }

  sendProps = () => ({
    ...this.props,
    ...this.state,
    onChange: this.onChange,
    onLoginSubmit: this.onLoginSubmit,
  })

  render() {
    return this.props.children(this.sendProps());
  }
}


const mapStateToProps = state => ({
  auth: state.auth.isAuthenticated,
  error: state.auth.error,
});

const matchDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      login,
    },
    dispatch,
  ),
});

export default connect(
  mapStateToProps,
  matchDispatchToProps,
)(AuthContainer);
