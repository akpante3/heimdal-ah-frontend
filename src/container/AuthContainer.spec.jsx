/* eslint-disable react/no-children-prop */
/* eslint-disable no-undef */
import React from 'react';
import { AuthContainer } from './AuthContainer';

describe('<AuthContainer />', () => {
  it('should render the AuthContainer component', () => {
    const wrapper = shallow(<AuthContainer children={() => {}} />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the AuthContainer component', () => {
    const wrapper = shallow(<AuthContainer children={() => {}} />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle change events', () => {
    const wrapper = shallow(<AuthContainer
      children={() => {}}
      auth={{ isAuthenticated: false, user: {}, error: {} }}
      error={{}}
      email='Heimdal'
      actions={() => {}}
    />);
    const e = { target: { name: 'email', value: 'Heimdal' }, preventDefault: () => {} };
    expect(wrapper.instance().props.email).toEqual('Heimdal');
    wrapper.instance().onChange(e);
    expect(wrapper.instance().props.email).toEqual(e.target.value);
  });

  it('calls `func` when mounted', () => {
    const wrapper = shallow(<AuthContainer
      children={() => {}}
      auth={{ isAuthenticated: false, user: {}, error: {} }}
      error={{}}
      email='Heimdal'
      actions={() => {}}
    />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'onLoginSubmit');
    const e = { target: { name: 'email', value: 'Heimdal' }, preventDefault: () => {} };
    wrapper.instance().onLoginSubmit(e);
    expect(instance.onLoginSubmit).toHaveBeenCalled();
  });
});
