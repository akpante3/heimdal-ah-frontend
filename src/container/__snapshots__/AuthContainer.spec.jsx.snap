// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthContainer /> should render the AuthContainer component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthContainer
    isLoading={false}
  >
    [Function]
  </AuthContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<AuthContainer /> should render the AuthContainer component 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthContainer
    isLoading={false}
  >
    [Function]
  </AuthContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
